weasyl/favorite.py

```diff
@@ -254,7 +254,27 @@ def check(userid, submitid=None, charid=None, journalid=None):
         ], options="bool")
 
 
-def count(submitid):
+def count(id, contenttype='submission'):
+    """Fetches the count of favorites on some content.
+
+    Args:
+        id (int): ID of the content to get the count for.
+        contenttype (str): Type of content to fetch. It accepts one of the following:
+            submission, journal, or character
+
+    Returns:
+        An int with the number of favorites.
+    """
+
+    if contenttype == 'submission':
+        querytype = 's'
+    elif contenttype == 'journal':
+        querytype = 'j'
+    elif contenttype == 'character':
+        querytype = 'f'
+    else:
+        raise ValueError("type should be one of 'submission', 'journal', or 'character'")
+
     return d.execute(
```

Comment:
As long as this is changing, can we move this to d.engine.execute() with named parameters?